packages/compiler/src/server/client-config-provider.ts

```diff
@@ -1,12 +1,18 @@
 import { Connection } from "vscode-languageserver/node.js";
+import { deepClone } from "../utils/misc.js";
 import { ServerHost } from "./types.js";
 
 interface LSPConfig {
   emit?: string[];
 }
 
+interface EntrypointConfig {
+  entrypoint?: string[];
+}
+
 interface Config {
   lsp?: LSPConfig;
+  compile?: EntrypointConfig;
```

Comment:
we have now 3 top level group `tsp-server`, `lsp`, `compile`, not sure this grouping make too much sense.

Shouldn't it just be at the root? Unrelated to this pr but `lsp` should probably have been combined with `tsp-server`

Reply:
yeah, i think we can combine the tsp-server and lsp, they are basically the same thing. as for entrypoint, i would prefer to keep the compile level, so in the future, when we are adding more other settings for compile which are not suitable at the root level, we won't hit the situation that some of compile settings are under compile while some are under root. thanks.

Reply:
My feeling was that from a user point of view it's not about compile it's about the extension knowing what to use as main file.
When I see emit option it could also very well be compile if we see it as what happens during compile.
Anyway not a major issue just a feeling as we had 3 groups for 3 options